### ISSUE TITLE
"Protect" a color

### ISSUE THREAD

**Issue by FlorinI** [drake7707/paintbynumbersgenerator#12]
Some colors are important for an image to the point of needing to be preserved as is and not averaged together with similar colors. For example, in the attached image the sand needs to stay yellowish rather than be turned into white.

On another note, @drake7707 please check your spambox for email from florin@----.----

![2020-08-25_20h03_39](https://user-images.githubusercontent.com/1222386/91205858-e83b9a00-e70e-11ea-86af-e352cca0a6b8.png)

**Comment by FlorinI** [comment]
The sand turns white even with a 30-color palette.
![paintbynumbers](https://user-images.githubusercontent.com/1222386/91206097-449eb980-e70f-11ea-9f03-1e411f4b4b7f.png)

**Comment by FlorinI** [comment]
In the same image, the shadow from the bottom right needs to be a darker shade from the flowers and the windows. This issue can be corrected in editing software like Illustrator because the shadow is a distinct area, but the sand is more difficult because it's merged into the house which is merged into the clouds.

I can think of a hack to correct even the sand (create a larger source image by adding a large area covered by that color, PBN that source image, then remove the added area from the result) but that's me working instead of the software :D

**Comment by FlorinI** [comment]
... and, after reading your comments on other issues, I will try playing with colorAliases. Will update here if I am able to unblock myself.

**Comment by FlorinI** [comment]
I wanted to share ths funny result I got when I tried to "protect" the sand's color. I entered "255,247,210" into "Restrict clustering colors" and I ended up with the genius image below.
![image](https://user-images.githubusercontent.com/1222386/92633657-c2e88780-f2db-11ea-9183-7a2177b0c6e9.png)
OK, now to find the good way to use this setting... :)

**Comment by FlorinI** [comment]
Same result when I tried to use the CLI with this setting:

![image](https://user-images.githubusercontent.com/1222386/92658198-d527ed00-f2fe-11ea-91c8-c36f270e3406.png)

It looks like if I specify one color, I have to specify all the colors.

I guess I could generate the PBN with the default settings and with 1 less color than I actually need, then copy the resulting palette + my color to the settings. Let me try this.

**Comment by FlorinI** [comment]
So I ran the CLI with this settings file:
`{
    "randomSeed": 0,
    "kMeansNrOfClusters": 25,
    "kMeansMinDeltaDifference": 1,
    "kMeansClusteringColorSpace": 0,
    "kMeansColorRestrictions": [	"sand-yellow",
									"color2-97-55",
									"color9-39-23",
									"color51-49-45",
									"color215-31-45",
									"color131-53-67",
									"color110-115-126",
									"color153-190-226",
									"color130-138-198",
									"color37-39-94",
									"color14-77-138",
									"color242-245-239",
									"color76-88-75",
									"color165-157-163",
									"color53-145-68",
									"color109-162-66",
									"color128-189-67",
									"color152-145-53",
									"color209-200-78",
									"color232-169-194",
									"color81-168-74",
									"color239-103-122",
									"color195-97-122",
									"color2-153-208",
									"color203-34-127"
 ],
    "colorAliases": {
        "A1": [            0,            0,            0        ],
        "A2": [            255,            0,            0        ],
        "A3": [            0,            255,            0        ],
        "A4": [            0,            0,            255        ],
        "B1": [            64,            64,            64        ],
        "B2": [            128,            128,            128        ],
        "B3": [            192,            192,            192        ],
        "sand-yellow": [            255,            247,            210        ],
		"color2-97-55": [      2,      97,      55    ],
		"color9-39-23": [      9,      39,      23    ],
		"color51-49-45": [      51,      49,      45    ],
		"color215-31-45": [      215,      31,      45    ],
		"color131-53-67": [      131,      53,      67    ],
		"color110-115-126": [      110,      115,      126    ],
		"color153-190-226": [      153,      190,      226    ],
		"color130-138-198": [      130,      138,      198    ],
		"color37-39-94": [      37,      39,      94    ],
		"color14-77-138": [      14,      77,      138    ],
		"color242-245-239": [      242,      245,      239    ],
		"color76-88-75": [      76,      88,      75    ],
		"color165-157-163": [      165,      157,      163    ],
		"color53-145-68": [      53,      145,      68    ],
		"color109-162-66": [      109,      162,      66    ],
		"color128-189-67": [      128,      189,      67    ],
		"color152-145-53": [      152,      145,      53    ],
		"color209-200-78": [      209,      200,      78    ],
		"color232-169-194": [      232,      169,      194    ],
		"color81-168-74": [      81,      168,      74    ],
		"color239-103-122": [      239,      103,      122    ],
		"color195-97-122": [      195,      97,      122    ],
		"color2-153-208": [      2,      153,      208    ],
		"color203-34-127": [      203,      34,      127    ]
    },
    "removeFacetsSmallerThanNrOfPoints": 20,
    "removeFacetsFromLargeToSmall": true,
    "maximumNumberOfFacets": 3000,
    "nrOfTimesToHalveBorderSegments": 2,
    "narrowPixelStripCleanupRuns": 3,
    "resizeImageIfTooLarge": true,
    "resizeImageWidth": 1024,
    "resizeImageHeight": 1024,
    "outputProfiles": [
        {
            "name": "full",
            "svgShowLabels": true,
            "svgFillFacets": true,
            "svgShowBorders": true,
            "svgSizeMultiplier": 1.5,
            "svgFontSize": 80,
            "svgFontColor": "#333",
            "filetype": "png"
        },
        {
            "name": "bordersLabels",
            "svgShowLabels": true,
            "svgFillFacets": false,
            "svgShowBorders": true,
            "svgSizeMultiplier": 1.5,
            "svgFontSize": 80,
            "svgFontColor": "#333",
            "filetype": "svg"
        },
        {
            "name": "jpgtest",
            "svgShowLabels": false,
            "svgFillFacets": true,
            "svgShowBorders": false,
            "svgSizeMultiplier": 1.5,
            "svgFontSize": 80,
            "svgFontColor": "#333",
            "filetype": "jpg",
            "filetypeQuality": 80
        }
    ]
}`
And I got back the exact same result as before, with 24 colors (which now have my aliases) - but no sand-yellow:
`[
  {
    "areaPercentage": 0.08684362584046455,
    "color": [
      2,
      97,
      55
    ],
    "colorAlias": "color2-97-55",
    "frequency": 72743,
    "index": 0
  },
  {
    "areaPercentage": 0.031217766274449877,
    "color": [
      9,
      39,
      23
    ],
    "colorAlias": "color9-39-23",
    "frequency": 26149,
    "index": 1
  },
  {
    "areaPercentage": 0.002430661674816626,
    "color": [
      51,
      49,
      45
    ],
    "colorAlias": "color51-49-45",
    "frequency": 2036,
    "index": 2
  },
  {
    "areaPercentage": 0.020061315709046453,
    "color": [
      215,
      31,
      45
    ],
    "colorAlias": "color215-31-45",
    "frequency": 16804,
    "index": 3
  },
  {
    "areaPercentage": 0.002997736476161369,
    "color": [
      131,
      53,
      67
    ],
    "colorAlias": "color131-53-67",
    "frequency": 2511,
    "index": 4
  },
  {
    "areaPercentage": 0.002581085727383863,
    "color": [
      110,
      115,
      126
    ],
    "colorAlias": "color110-115-126",
    "frequency": 2162,
    "index": 5
  },
  {
    "areaPercentage": 0.17755768643031786,
    "color": [
      153,
      190,
      226
    ],
    "colorAlias": "color153-190-226",
    "frequency": 148728,
    "index": 6
  },
  {
    "areaPercentage": 0.047739341381418096,
    "color": [
      130,
      138,
      198
    ],
    "colorAlias": "color130-138-198",
    "frequency": 39988,
    "index": 7
  },
  {
    "areaPercentage": 0.01622789005195599,
    "color": [
      37,
      39,
      94
    ],
    "colorAlias": "color37-39-94",
    "frequency": 13593,
    "index": 8
  },
  {
    "areaPercentage": 0.04942504584352078,
    "color": [
      14,
      77,
      138
    ],
    "colorAlias": "color14-77-138",
    "frequency": 41400,
    "index": 9
  },
  {
    "areaPercentage": 0.20015472188264058,
    "color": [
      242,
      245,
      239
    ],
    "colorAlias": "color242-245-239",
    "frequency": 167656,
    "index": 10
  },
  {
    "areaPercentage": 0.0025082613844743277,
    "color": [
      76,
      88,
      75
    ],
    "colorAlias": "color76-88-75",
    "frequency": 2101,
    "index": 11
  },
  {
    "areaPercentage": 0.0024127540495110025,
    "color": [
      165,
      157,
      163
    ],
    "colorAlias": "color165-157-163",
    "frequency": 2021,
    "index": 12
  },
  {
    "areaPercentage": 0.06668919047982885,
    "color": [
      53,
      145,
      68
    ],
    "colorAlias": "color53-145-68",
    "frequency": 55861,
    "index": 13
  },
  {
    "areaPercentage": 0.016541870415647923,
    "color": [
      109,
      162,
      66
    ],
    "colorAlias": "color109-162-66",
    "frequency": 13856,
    "index": 14
  },
  {
    "areaPercentage": 0.05412281288202934,
    "color": [
      128,
      189,
      67
    ],
    "colorAlias": "color128-189-67",
    "frequency": 45335,
    "index": 15
  },
  {
    "areaPercentage": 0.007214385314792176,
    "color": [
      152,
      145,
      53
    ],
    "colorAlias": "color152-145-53",
    "frequency": 6043,
    "index": 16
  },
  {
    "areaPercentage": 0.016170585650977995,
    "color": [
      209,
      200,
      78
    ],
    "colorAlias": "color209-200-78",
    "frequency": 13545,
    "index": 17
  },
  {
    "areaPercentage": 0.058641503667481665,
    "color": [
      232,
      169,
      194
    ],
    "colorAlias": "color232-169-194",
    "frequency": 49120,
    "index": 18
  },
  {
    "areaPercentage": 0.00782085689180929,
    "color": [
      81,
      168,
      74
    ],
    "colorAlias": "color81-168-74",
    "frequency": 6551,
    "index": 19
  },
  {
    "areaPercentage": 0.0588062538202934,
    "color": [
      239,
      103,
      122
    ],
    "colorAlias": "color239-103-122",
    "frequency": 49258,
    "index": 20
  },
  {
    "areaPercentage": 0.007856672142420537,
    "color": [
      195,
      97,
      122
    ],
    "colorAlias": "color195-97-122",
    "frequency": 6581,
    "index": 21
  },
  {
    "areaPercentage": 0.021831782930929095,
    "color": [
      2,
      153,
      208
    ],
    "colorAlias": "color2-153-208",
    "frequency": 18287,
    "index": 22
  },
  {
    "areaPercentage": 0.04214619307762836,
    "color": [
      203,
      34,
      127
    ],
    "colorAlias": "color203-34-127",
    "frequency": 35303,
    "index": 23
  }
]`
What am I doing wrong?
